### ISSUE TITLE
keras2.0.2 fit_generator have some problem

### ISSUE THREAD

**Issue by SpenserCai** [fchollet/keras#6117]
My code is :

import glob
import ConfigParser
import numpy as np
from scipy import misc
from keras.layers import Input, Conv2D, MaxPooling2D, Flatten, Activation, Dense, Dropout, ZeroPadding2D
from keras.models import Model
from keras.utils.np_utils import to_categorical
from tqdm import tqdm
import sys
...
imgsize = (252L, 96L, 3L)
batch_size = int(getConfig("train","batch_size")) # 256 Every batch's size
vcodelen = int(getConfig("train","vcodelen")) # 4 Vercode's length
nb_class = int(getConfig("train","nb_class")) # 43 Number of classification
imgs = glob.glob('sample/*.jpg')
...
model = Model(input=inputs, output=[Dense(nb_class, activation='softmax')(cnn) for i in range(vcodelen)])
...
nb_epoch = int(sys.argv[2])
model.fit_generator(generate_best(), steps_per_epoch=len(imgs), epochs=nb_epoch)

in keras 1.2.2:
2017-04-03 09:50:44.856965: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0 
2017-04-03 09:50:44.856972: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y 
2017-04-03 09:50:44.856982: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1070, pci bus id: 0000:03:00.0)
128/100300 [..............................] - ETA: 4842s - loss: 17.2639 - dense_2_loss: 4.3186 - dense_3_loss: 4.3175 - dense_4_loss: 4.3175   
256/100300 [..............................] - ETA: 2715s - loss: 16.3546 - dense_2_loss: 4.1353 - dense_3_loss: 4.0899 - dense_4_loss: 4.0707   
384/100300 [..............................] - ETA: 2006s - loss: 29.3421 - dense_2_loss: 7.3398 - dense_3_loss: 6.9857 - dense_4_loss: 7.6607   
512/100300 [..............................] - ETA: 1651s - loss: 26.4200 - dense_2_loss: 6.6322 - dense_3_loss: 6.3918 - dense_4_loss: 6.7715   
640/100300 [..............................] - ETA: 1438s - loss: 24.4368 - dense_2_loss: 6.1310 - dense_3_loss: 5.9369 - dense_4_loss: 6.2451   
768/100300 [..............................] - ETA: 1295s - loss: 23.2084 - dense_2_loss: 5.8215 - dense_3_loss: 5.6585 - dense_4_loss: 5.9153   
896/100300 [..............................] - ETA: 1193s - loss: 22.3293 - dense_2_loss: 5.5988 - dense_3_loss: 5.4595 - dense_4_loss: 5.6792  
1024/100300 [..............................] - ETA: 1116s - loss: 21.6555 - dense_2_loss: 5.4287 - dense_3_loss: 5.3059 - dense_4_loss: 5.4990

but in keras2.0.2:
2017-04-03 10:13:39.950290: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1070, pci bus id: 0000:03:00.0)
1/100300 [..............................] - ETA: 613333s - loss: 17.2686 - dense_2_loss: 4.3167 - dense_3_loss: 4.3177 - dense_4_loss: 4.31     
2/100300 [..............................] - ETA: 345075s - loss: 16.4263 - dense_2_loss: 4.1212 - dense_3_loss: 4.1022 - dense_4_loss: 4.10     
3/100300 [..............................] - ETA: 255598s - loss: 28.3589 - dense_2_loss: 6.6864 - dense_3_loss: 6.8787 - dense_4_loss: 7.43     
4/100300 [..............................] - ETA: 210892s - loss: 26.0758 - dense_2_loss: 6.0743 - dense_3_loss: 6.4762 - dense_4_loss: 6.73     
5/100300 [..............................] - ETA: 184153s - loss: 24.1586 - dense_2_loss: 5.6787 - dense_3_loss: 6.0037 - dense_4_loss: 6.21

I don't know why?

**Comment by myutwo150** [comment]
Try to divide `steps_per_epoch` by `batch_size`

**Comment by SpenserCai** [comment]
oh maybe Keras1 is sample_per_epoch, but now is steps_per_epoch,so I think this way need write sample/batch_size(e.g. 100300/128+1)

**Comment by royxue** [comment]
You dont have to plus 1, during training, Keras will automatically do that for you.

**Comment by MarxSoul55** [comment]
@myutwo150 is correct. You have to account for `steps_per_epoch` as per the [documentation](https://keras.io/models/sequential/#fit_generator). Otherwise you'll get insane ETAs (just tested it out myself).
